### PR TITLE
CNF-15750: Cleanup inventory API and code generate server

### PR DESCRIPTION
**Pull request by alegacy**
This commit is the beginning of aligning the inventory (resource) API endpoints with the new implementation of the alarm server.  In order to reduce boilerplate code the inventory (resource) server structure will now be code generated from the openapi yaml.  Since certain API components (e.g.,  parameter definitions) are common to both the alarm and inventory components they are split out into a "common" package.

### CODE REVIEW

File: internal/service/alarms/api/openapi.yaml [hunk 1]
```diff
@@ -34,13 +34,9 @@ paths:
...
-        - in: query
-          name: filter
-          schema:
-            type: string
-          allowReserved: true
-          description: Optional filter parameter. Only the attributes in AlarmEventRecord are filterable.
-          example: (eq,perceivedSeverity,0)
+      - $ref: "../../common/api/openapi.yaml#/components/parameters/excludeFields"
+      - $ref: "../../common/api/openapi.yaml#/components/parameters/fields"
+      - $ref: "../../common/api/openapi.yaml#/components/parameters/filter"
```
Comment (pixelsoccupied) [inline internal/service/alarms/api/openapi.yaml:39]: Ah we totally missed the other params from spec...I only noticed `filter` param for alarms (cc @mlguerrero12)

**Review by pixelsoccupied — COMMENTED** [review]
Looks good!

File: internal/service/alarms/api/openapi.yaml [hunk 2]
```diff
@@ -34,13 +34,9 @@ paths:
       tags:
         - alarms
       parameters:
-        - in: query
-          name: filter
```
[the remark this answers is not in the document]
Comment (alegacy) [reply internal/service/alarms/api/openapi.yaml:38]: Yes, good point.  I'll do that.

**Pull request merged**